versioned_hdf5/wrappers.py

```diff
@@ -634,30 +635,52 @@ def resize(self, size, axis=None):
 
         old_shape = self.shape
         data_dict = self.id.data_dict
+        can_read_direct = self.id.can_read_direct
 
         old_shape_idx = Tuple(*[Slice(0, i) for i in old_shape])
         new_data_dict = {}
-        for c in self.chunks.as_subchunks(old_shape_idx, size):
-            if c in data_dict:
-                new_data_dict[c] = data_dict[c]
+        for chunk in self.chunks.as_subchunks(old_shape_idx, size):
+            if chunk in data_dict:
+                new_data_dict[chunk] = data_dict[chunk]
             else:
-                a = self[c.raw]
-                data = np.full(c.newshape(size), self.fillvalue, dtype=self.dtype)
-                index = old_shape_idx.as_subindex(c)
-                data[index.raw] = a
-                new_data_dict[c] = data
+                data = np.full(chunk.newshape(size), self.fillvalue, dtype=self.dtype)
+                index = old_shape_idx.as_subindex(chunk)
+                data[index.raw] = self.get_index(
+                    chunk.raw, can_read_direct=can_read_direct
+                )
+                new_data_dict[chunk] = data
 
         self.id.data_dict = new_data_dict
         self.id.shape = size
 
     @with_phil
-    def __getitem__(self, args, new_dtype=None):
-        """Read a slice from the HDF5 dataset.
+    def get_index(
+        self,
+        args: Union[slice, Slice, Tuple, tuple, h5r.RegionReference],
+        new_dtype: Optional[str] = None,
+        can_read_direct: bool = True,
```

Comment:
I think the default should be `None` and if it's not passed we default to evaluating `self.id.can_read_direct`.

Reply:
Sounds good, I'll make the change.